core/src/main/java/org/ehcache/core/EhcacheBase.java

```diff
@@ -118,53 +119,59 @@
   @Override
   public V get(K key) {
     getObserver.begin();
-    statusTransitioner.checkAvailable();
-    checkNonNull(key);
-
     try {
-      Store.ValueHolder<V> valueHolder = doGet(key);
+      statusTransitioner.checkAvailable();
+      checkNonNull(key);
 
-      // Check for expiry first
-      if (valueHolder == null) {
-        getObserver.end(GetOutcome.MISS);
-        return null;
-      } else {
-        getObserver.end(GetOutcome.HIT);
-        return valueHolder.get();
-      }
-    } catch (StoreAccessException e) {
       try {
-        return resilienceStrategy.getFailure(key, e);
-      } finally {
+        Store.ValueHolder<V> valueHolder = doGet(key);
+
+        // Check for expiry first
+        if (valueHolder == null) {
+          getObserver.end(GetOutcome.MISS);
+          return null;
+        } else {
+          getObserver.end(GetOutcome.HIT);
+          return valueHolder.get();
+        }
+      } catch (StoreAccessException e) {
+        V value = resilienceStrategy.getFailure(key, e);
         getObserver.end(GetOutcome.FAILURE);
+        return value;
       }
+    } catch(Exception e) {
```

Comment:
This should be `} catch (Throwable) {` right?

Reply:
I always feel not at ease when catching throwable... But since we really want to finish the thing, I guess you are right.